### PR TITLE
python3Packages.Pyro4: Fix tests on darwin

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — APPROVED** [review]

File: pkgs/development/python-modules/pyro4/default.nix [hunk 1]
```diff
@@ -32,10 +32,15 @@ buildPythonPackage rec {
...
   checkInputs = [ pytestCheckHook ];
+
   # add testsupport.py to PATH
   preCheck = "PYTHONPATH=tests/PyroTests:$PYTHONPATH";
+
   # ignore network related tests, which fail in sandbox
-  pytestFlagsArray = [ "--ignore=tests/PyroTests/test_naming.py" ];
+  pytestFlagsArray = [ "--ignore=tests/PyroTests/test_naming.py" ]
+  # test hangs on darwin with sandbox enabled
+    ++ lib.optionals stdenv.isDarwin [ "--ignore=PyroTests/test_daemon.py" ];
```
Comment (prusnak) [inline pkgs/development/python-modules/pyro4/default.nix:42]: Maybe it is possible to use `disabledTests` instead?

```suggestion
  disabledTests = [
    "test_naming"
  ] ++ lib.optionals stdenv.isDarwin [
    "test_daemon"
  ];
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/pyro4/default.nix:42]: disabledTests can only disable single tests but we are ignoring entire files here with multiple tests. See https://github.com/irmen/Pyro4/blob/4.80/tests/PyroTests/test_daemon.py

**Comment by bhipple** [comment]
Result of `nixpkgs-review pr 108253` [1](https://github.com/Mic92/nixpkgs-review)

**Review by bhipple — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This wasn't actually ready as the nixpkgs-review still hangs.

**Pull request merged**